### PR TITLE
Build images with deps versions

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

Since the dependencies are getting statically linked into the binary artifacts that we release, we need to have a good/easy way to upgrade the versions of such dependencies and make sure they are consistent for all the different builds.

### Modifications

 1. Use a single `dependencies.yaml` file to declare the dependencies
 2. The versions are then converted to variable inside the Dockerfiles
 3. Create (and cache) the Docker images used for building in the Github actions job

### CODE REVIEW

File: .github/workflows/ci-build-binary-artifacts.yaml [hunk 1]
```diff
@@ -43,8 +41,20 @@ jobs:
...
         run: build-support/generate-source-archive.sh
 
+      - uses: docker/setup-buildx-action@v2
+      - run: build-support/copy-deps-versionfile.sh
+
+      - name: Build dependencies Docker image
+        uses: docker/build-push-action@v3
+        with:
+          context: ./pkg/deb
+          tags: ${{ env.TEST_TAG }}
```
Comment (BewareMyPower) [inline .github/workflows/ci-build-binary-artifacts.yaml:51]: Could you explain where the `TEST_TAG` environment variable is set?

File: pkg/deb/Dockerfile [hunk 2]
```diff
@@ -25,69 +25,93 @@ ARG PLATFORM
...
-RUN curl -O -L https://github.com/openssl/openssl/archive/OpenSSL_1_1_1n.tar.gz && \
-    tar xfz OpenSSL_1_1_1n.tar.gz && \
-    cd openssl-OpenSSL_1_1_1n/ && \
+RUN SNAPPY_VERSION=$(dep-version.py snappy) && \
+    curl -O -L https://github.com/google/snappy/archive/refs/tags/${SNAPPY_VERSION}.tar.gz && \
+    tar xfz ${SNAPPY_VERSION}.tar.gz && \
+    cd snappy-${SNAPPY_VERSION} && \
+    CXXFLAGS="-fPIC -O3" cmake . -DSNAPPY_BUILD_TESTS=OFF -DSNAPPY_BUILD_BENCHMARKS=OFF && \
+    make -j8 && make install && \
+    rm -rf /snappy-${SNAPPY_VERSION} /snappy-${SNAPPY_VERSION}.tar.gz
```
Comment (BewareMyPower) [inline pkg/deb/Dockerfile:93]:
```suggestion
    rm -rf /snappy-${SNAPPY_VERSION} /${SNAPPY_VERSION}.tar.gz
```

File: pkg/apk/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    rm -rf /zstd-${ZSTD_VERSION} /zstd-${ZSTD_VERSION}.tar.gz
+
+# Snappy
+RUN SNAPPY_VERSION=$(dep-version.py snappy) && \
+    curl -O -L https://github.com/google/snappy/archive/refs/tags/${SNAPPY_VERSION}.tar.gz && \
+    tar xfz ${SNAPPY_VERSION}.tar.gz && \
+    cd snappy-${SNAPPY_VERSION} && \
+    CXXFLAGS="-fPIC -O3" cmake . -DSNAPPY_BUILD_TESTS=OFF -DSNAPPY_BUILD_BENCHMARKS=OFF && \
+    make -j8 && make install && \
+    rm -rf /snappy-${SNAPPY_VERSION} /snappy-${SNAPPY_VERSION}.tar.gz
```
Comment (BewareMyPower) [inline pkg/apk/Dockerfile:88]:
```suggestion
    rm -rf /snappy-${SNAPPY_VERSION} /${SNAPPY_VERSION}.tar.gz
```

File: pkg/rpm/Dockerfile [hunk 4]
```diff
@@ -24,71 +24,94 @@ FROM centos:7
...
-RUN curl -O -L https://github.com/openssl/openssl/archive/OpenSSL_1_1_1n.tar.gz && \
-    tar xfz OpenSSL_1_1_1n.tar.gz && \
-    cd openssl-OpenSSL_1_1_1n/ && \
+RUN SNAPPY_VERSION=$(dep-version.py snappy) && \
+    curl -O -L https://github.com/google/snappy/archive/refs/tags/${SNAPPY_VERSION}.tar.gz && \
+    tar xfz ${SNAPPY_VERSION}.tar.gz && \
+    cd snappy-${SNAPPY_VERSION} && \
+    CXXFLAGS="-fPIC -O3" cmake . -DSNAPPY_BUILD_TESTS=OFF -DSNAPPY_BUILD_BENCHMARKS=OFF && \
+    make -j8 && make install && \
+    rm -rf /snappy-${SNAPPY_VERSION} /snappy-${SNAPPY_VERSION}.tar.gz
```
Comment (BewareMyPower) [inline pkg/rpm/Dockerfile:94]:
```suggestion
    rm -rf /snappy-${SNAPPY_VERSION} /${SNAPPY_VERSION}.tar.gz
```

**Review by BewareMyPower — CHANGES_REQUESTED** [review]
The snappy tar file downloaded is `<version>.tar.gz`.

**Review by BewareMyPower — APPROVED** [review]

File: .github/workflows/ci-build-binary-artifacts.yaml (hunk 1 above)
Comment (merlimat) [reply .github/workflows/ci-build-binary-artifacts.yaml:51]: That’s a good point and a good catch. I thought it was auto assigned some unique tag, though it just ends up being an empty tag, so it goes using the default image, which is already on dockerhub. I’ll fix it

**Pull request merged**